spec/integration/users/user_show_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+RSpec.describe 'userpostpage#show', type: :feature do
+  before do
+    @user = User.create(name: 'Afimia Fidelis',
+                        photo: 'https://res.cloudinary.com/dsfuiu63q/image/upload/v1678870006/avatars/icon-256x256_wa30f0.png',
+                        bio: 'Fullstack Dev Freelance',
+                        posts_count: 0)
+    @post1 = Post.create(author_id: @user.id, title: 'My First Post', text: 'This is the content of my first post.',
+                         likes_count: 0, comments_count: 0)
+    @post2 = Post.create(author_id: @user.id, title: 'Rails Rocks', text: 'Excited about Rails development!',
+                         likes_count: 0, comments_count: 0)
+    @post3 = Post.create(author_id: @user.id, title: 'Learning Ruby', text: 'Exploring the beauty of Ruby language.',
+                         likes_count: 0, comments_count: 0)
+    visit user_path(@user.id)
+  end
+
+  it 'I can see the user\'s profile picture' do
+    expect(page).to have_css("img[src*='https://res.cloudinary.com/dsfuiu63q/image/upload/v1678870006/avatars/icon-256x256_wa30f0.png']")
+  end
+
+  it 'I can see the user\'s profile name' do
+    expect(page).to have_content 'Afimia Fidelis'
+  end
+
+  it 'I can see the number of posts the user has written' do
+    expect(page).to have_content 'Number of Posts: 3'
+  end
+
+  it 'I can see the user\'s bio' do
+    expect(page).to have_content 'Fullstack Dev Freelance'
+  end
+
+  it 'I can see the user\'s first 3 posts' do
+    expect(page).to have_content('This is the content of my first post.')
+    expect(page).to have_content('Excited about Rails development!')
+    expect(page).to have_content('Exploring the beauty of Ruby language.')
+  end
```

Comment:
👍 Good progress on your project! To ensure comprehensive testing, consider the following suggestion:

- [ ] **User's First 3 Posts Test:** While your current test case checks for the existence of a user's first three posts, it's beneficial to enhance the coverage. Add atlest 4 posts to the user and verify that the displayed page includes the first three posts and excludes any posts beyond that.

   ```ruby
   # Example Test Case

     it 'displays the first 3 posts and excludes others do
      ...................
       expect(page).not_to have_content(posts.last.title) #fourth post checking
   end
   ```

   This test case creates a user and four posts, then checks that the user's show page displays the first three posts and excludes the last post. Adjust the test case based on your application's specifics.

By expanding the test scenario to include more posts, you'll ensure that your code adheres to the project requirements and handles a larger set of data. Keep up the great work! 🚀🧪